config/webpack.config.js

```diff
@@ -118,75 +122,159 @@ if (!__TEST__) {
 // Loaders
 // ------------------------------------
 // JavaScript / JSON
-webpackConfig.module.loaders = [{
+webpackConfig.module.rules = [{
   test    : /\.(js|jsx)$/,
   exclude : /node_modules/,
-  loader  : 'babel',
+  loader  : 'babel-loader',
   query   : project.compiler_babel
 }, {
   test   : /\.json$/,
-  loader : 'json'
+  loader : 'json-loader'
 }]
 
 // ------------------------------------
 // Style Loaders
 // ------------------------------------
 // We use cssnano with the postcss loader, so we tell
 // css-loader not to duplicate minimization.
-const BASE_CSS_LOADER = 'css?sourceMap&-minimize'
-
-webpackConfig.module.loaders.push({
-  test    : /\.scss$/,
-  exclude : null,
-  loaders : [
-    'style',
-    BASE_CSS_LOADER,
-    'postcss',
-    'sass?sourceMap'
+webpackConfig.module.rules.push({
+  test : /\.scss$/,
+  use  : [
+    {
+      loader: 'style-loader'
+    },
+    {
+      loader      : 'css-loader',
+      options     : {
+        sourceMap : true,
+        minimize  : true
+      }
+    },
+    {
+      loader : 'postcss-loader'
+    },
+    {
+      loader  : 'sass-loader',
+      options : {
+        sourceMap : true
+      }
+    }
   ]
 })
-webpackConfig.module.loaders.push({
-  test    : /\.css$/,
-  exclude : null,
-  loaders : [
-    'style',
-    BASE_CSS_LOADER,
-    'postcss'
+webpackConfig.module.rules.push({
+  test : /\.css$/,
+  use  : [
+    {
+      loader : 'style-loader'
+    },
+    {
+      loader  : 'css-loader',
+      options : {
+        sourceMap : true,
+        minimize  : true
+      }
+    },
+    {
+      loader : 'postcss-loader'
+    }
   ]
 })
 
-webpackConfig.sassLoader = {
-  includePaths : project.paths.client('styles')
-}
-
-webpackConfig.postcss = [
-  cssnano({
-    autoprefixer : {
-      add      : true,
-      remove   : true,
-      browsers : ['last 2 versions']
-    },
-    discardComments : {
-      removeAll : true
-    },
-    discardUnused : false,
-    mergeIdents   : false,
-    reduceIdents  : false,
-    safe          : true,
-    sourcemap     : true
+webpackConfig.plugins.push(
+  new webpack.LoaderOptionsPlugin({
+    options : {
+      postcss : [
+        cssnano({
+          autoprefixer : {
+            add      : true,
+            remove   : true,
+            browsers : ['last 2 versions']
+          },
+          discardComments: {
+            removeAll : true
+          },
+          discardUnused : false,
+          mergeIdents   : false,
+          reduceIdents  : false,
+          safe          : true,
+          sourcemap     : true
+        })
+      ],
+      sassLoader : {
+        includePaths : project.paths.client('styles')
+      }
+    }
   })
-]
+)
 
 // File loaders
 /* eslint-disable */
```

Comment:
Less boilerplate:

    const FONT_EXTS = Map([
      'woff', 'application/font-woff',
      'woff2', 'application/font-woff2',
      'otf', 'font/opentype',
      'ttf', 'application/octet-stream',
      'eot', 'application/vnd.ms-fontobject',
      'svg', 'image/svg+xml',
    ]);
    for (let [extension, mimeType] of FONT_EXTS) {
      webpackConfig.module.rules.push({
        test    : `\\.${extension}(\\?.*)?$`,
        loader  : 'url-loader',
        options : {
          prefix   : 'fonts/',
          name     : '[path][name].[ext]',
          limit    : '10000',
          mimetype : mimeType,
        },
      });
    };
    webpackConfig.module.rules.push(
      {
        test    : /\.(png|jpg|gif)$/,
        loader  : 'url-loader',
        options : {
          limit    : '8192',
        },
      },
    );

(Note addition of .gif)
I haven't tested this code yet.